edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Agent.cs

```diff
@@ -131,7 +133,8 @@ public async Task ReconcileAsync(CancellationToken token)
                     else
                     {
                         ModuleSet desiredModuleSet = deploymentConfig.GetModuleSet();
-                        this.availabilityMetric.ComputeAvailability(desiredModuleSet, current);
+                        _ = Task.Run(() => this.availabilityMetric.ComputeAvailability(desiredModuleSet, current))
+                            .ContinueWith(t => Log.LogError($"ComputeAvailability failed. Reason: {t.Exception}"), TaskContinuationOptions.OnlyOnFaulted);
```

Comment:
Please read this, https://stackoverflow.com/questions/5613951/simplest-way-to-do-a-fire-and-forget-method-in-c-sharp-4-0

Reply:
added .ConfigureAwait(false)